lib/converter.js

```diff
@@ -1,20 +1,38 @@
 'use strict';
 
-function dictionary(obj) {
-  const dictionaryHelper = (obj, accumulator, propPath) => {
-    Object.keys(obj).forEach(key => {
-      const value = obj[key];
-      const newPropPath = propPath ? `${propPath}${this.propSeparator}${key}` : key;
-
-      if (this.excludeProps.includes(newPropPath)) return;
-      if (value instanceof Object) return dictionaryHelper(value, accumulator, newPropPath);
-      accumulator[newPropPath] = value;
-    });
+function convert(data, options) {
+  const { excludeHeader, delimiter } = options;
+  const arrData = !Array.isArray(data) ? [data] : data;
+  const processedData = unwind(arrData, options);
+  return transformToRows(processedData, delimiter, excludeHeader);
+}
+
+module.exports = { convert };
+
+function unwind(data, options) {
+  return data.map(it => {
+    const flatten = (obj, accumulator, propPath) => {
+      Object.keys(obj).forEach(key => {
+        const value = obj[key];
+        const newPropPath = getNewPropPath(propPath, key, options);
 
-    return accumulator;
-  };
+        if (!newPropPath) return;
+        if (value instanceof Object) return flatten(value, accumulator, newPropPath);
+        accumulator[newPropPath] = value;
+      });
 
-  return dictionaryHelper(obj, {});
+      return accumulator;
+    };
+
+    return flatten(it, {});
+  });
+}
+
+function getNewPropPath(currentPath, key, opts) {
+  const testPath = currentPath ? `${currentPath}${opts.propSeparator}${key}` : key;
```

Comment:
Why is this called test path?

Reply:
Because it's evaluated should it be excluded or renamed, so I thought it fitted in. An alternative could be newPropPath but when I thought of it earlier the testPath fitted better and it made line 32 shorter so I could fit everything into a single line instead of needing to break it.

Another alternative is path. What do you think?

Reply:
This has been fixed in PR #20